sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/calciteCompiler/CustomFunctions.java

```diff
@@ -0,0 +1,227 @@
+package org.dbsp.sqlCompiler.compiler.frontend.calciteCompiler;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeField;
+import org.apache.calcite.sql.SqlCallBinding;
+import org.apache.calcite.sql.SqlFunction;
+import org.apache.calcite.sql.SqlFunctionCategory;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlOperandCountRange;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.SqlOperatorBinding;
+import org.apache.calcite.sql.type.InferTypes;
+import org.apache.calcite.sql.type.OperandTypes;
+import org.apache.calcite.sql.type.ReturnTypes;
+import org.apache.calcite.sql.type.SqlOperandCountRanges;
+import org.apache.calcite.sql.type.SqlOperandTypeChecker;
+import org.apache.calcite.sql.type.SqlOperandTypeInference;
+import org.apache.calcite.sql.type.SqlReturnTypeInference;
+import org.apache.calcite.sql.type.SqlSingleOperandTypeChecker;
+import org.apache.calcite.sql.type.SqlTypeFamily;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.dbsp.sqlCompiler.compiler.errors.CompilationError;
+import org.dbsp.sqlCompiler.compiler.frontend.CalciteObject;
+import org.dbsp.util.Linq;
+import org.dbsp.util.Utilities;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+import static org.apache.calcite.sql.type.OperandTypes.family;
+import static org.apache.calcite.sql.type.ReturnTypes.ARG1;
+
+/** Several functions that we define and add to the existing ones. */
+public class CustomFunctions {
+    private final List<SqlFunction> initial = new ArrayList<>();
+    private final HashMap<String, ExternalFunction> udf;
+
+    public CustomFunctions() {
+        this.initial.add(new SqlDivisionFunction());
+        this.initial.add(new RlikeFunction());
+        this.initial.add(new WriteLogFunction());
+        this.udf = new HashMap<>();
+    }
+
+    static class RlikeFunction extends SqlFunction {
```

Comment:
The static classes here were moved from CalciteCompiler.

Reply:
Shouldn't we be attributing them?